odoo/addons/base/models/ir_model.py

```diff
@@ -2212,7 +2213,8 @@ def _process_end(self, modules):
             if record.exists():
                 module = xmlid.split('.', 1)[0]
                 record = record.with_context(module=module)
-                if record._name == 'ir.model.fields' and not module.startswith('test_'):
+                if record._name == 'ir.model.fields' and not module.startswith('test_') \
+                    and "field:%s.%s" % (record.model, record.name,) not in os.environ.get('suppress_upgrade_warnings', '').split(','):
```

Comment:
You can cache the environment variable when loading the module:
```python
SUPPRESS_UPGRADE_WARNINGS = set(os.environ.get('suppress_upgrade_warnings', '').split(','))
```
which makes the following faster
```suggestion
                    and "field:%s.%s" % (record.model, record.name,) not in SUPPRESS_UPGRADE_WARNINGS:
```